packages/pwa-buildpack/lib/Utilities/addImgOptMiddleware.js

```diff
@@ -59,30 +63,18 @@ If possible, install additional tools to build NodeJS native dependencies:
 https://github.com/nodejs/node-gyp#installation`
         );
     } else {
-        const toExpressSharpUrl = (incomingUrl, incomingQuery) => {
-            const imageUrl = new URL(incomingUrl, backendUrl);
-            debug('imageUrl', imageUrl);
-
-            const optParamNames = ['auto', 'format', 'width', 'height'];
-
-            // Start with the original search params, so
-            // we can preserve any non-imageopt parameters
-            // others might want.
-            const params = new URLSearchParams(imageUrl.search);
-            for (const param of optParamNames) {
-                params.delete(param);
-            }
-            params.set('url', imageUrl.pathname);
-            if (incomingQuery.format === 'pjpg') {
-                params.set('progressive', 'true');
+        const toExpressSharpQuery = query => {
+            // Venia's makeUrl expects the Fastly style of imageopto params.
+            // https://docs.fastly.com/api/imageopto/
+            // Rewrite them just a little bit.
+            if (query.format === 'pjpg') {
+                query.progressive = true;
+                query.format = 'jpeg';
             }
-            if (incomingQuery.auto === 'webp') {
-                params.set('format', 'webp');
+            if (query.auto === 'webp') {
+                query.format = 'webp';
             }
```

Comment:
I think we should support .png as well. We know that .png doesn't perform well on the web but in production shops there is a very likely possibility that merchants will use this format (just like we do in our current project). Best we can do is set .png to progressive just like we do for .jpeg

Reply:
I haven't tested that! I believe this supports PNG; all you have to do is use URL parameters `format=png&progressive=true`. By default, it'll optimize PNGs into WebP files, which I think is a good default, but that query string should override it.

Reply:
That's correct, but ios & safari don't support WebP. So till they do i believe we should make a .png case

Reply:
Yea, I think we need to fix this on Safari.